src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/QueryAssignmentIntegrationTest.java

```diff
@@ -159,6 +162,46 @@ public void retrieveRoleAssignmentsByQueryRequest_queryRequestWithoutHeaders() t
             .andReturn();
     }
 
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_role_assignment.sql"})
+    public void retrieveRoleAssignmentsByQueryRequest_verifyRoleLabel() throws Exception {
+        retrieveRoleAssignmentsByQueryRequest_conditionallyAddRoleLabel(true);
+        retrieveRoleAssignmentsByQueryRequest_conditionallyAddRoleLabel(false);
+    }
+
+    void retrieveRoleAssignmentsByQueryRequest_conditionallyAddRoleLabel(Boolean includeLabels) throws Exception {
+
+        logger.info("Retrieve Role Assignments when includeLabels {}", includeLabels);
+
+        final MvcResult result = mockMvc.perform(post(URL)
+                                                     .contentType(JSON_CONTENT_TYPE)
+                                                     .headers(getHttpHeaders("3", "roleName"))
+                                                     .param(INCLUDE_LABELS_PARAM, includeLabels.toString())
+                                                     .content(mapper.writeValueAsBytes(
+                                                         QueryRequest.builder()
+                                                             .actorId(ACTOR_ID_2).build())))
+            .andExpect(status().isOk())
+            .andReturn();
+        JsonNode responseJsonNode = new ObjectMapper().readValue(result.getResponse().getContentAsString(),
```

Comment:
Lines 185 to 202 and 317 to 334 contain similar code. This can be refactored into a method to avoid duplication.

Reply:
thanks for feedback, see [3322b75](https://github.com/hmcts/am-role-assignment-service/pull/2355/commits/3322b7539e138b7f2ae7d262b0719201b7fd89be) for refactor

Reply:
This has been re-reviewed. 👍